### PR TITLE
[Feature] Added automatic creation of talking point forum post

### PR DESCRIPTION
**Pull request by ThePhysic**
## Summary
What is the context behind this change? (e.g. background/problem it’s solving)
- It takes extra time and effort to create a forum post from something you said in chat every time, especially if you do this often

What does this change do? (ex. high-level impact)
- Saying "Talking point" or "TP" in the chat will automatically create a post in a forum channel, containing the content from your message

What do each of the code-blocks/functions in this PR do? (ex. low-level description)
- Uses regex to check for keyword in new messages
- Discord Butler will prompt you for a title for the post, activating a collector to filter response
- Creates post in forum channel with embed of original message
- Sends message in chat with link to new forum post

## Testing
**Automated Testing:**
N/A

**Manual Testing:**
[Forum Post](https://github.com/ThePhysic/Discord-Butler/assets/57155067/ab4d97da-28b2-4fc8-80e6-fada8969dfd2)
[Link in chat to forum post](https://github.com/ThePhysic/Discord-Butler/assets/57155067/362b248b-014b-47b1-af10-3b9cd9a1494b)

## References
[Discord.js](https://discord.js.org/docs/packages/discord.js/14.15.3/Collector:Class)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -33,9 +33,11 @@ const TALKING_POINT_TARGET_CHANNEL_ID = process.env.TALKING_POINT_CHANNEL_ID;
 const GENERAL_CHANNEL_ID = process.env.GENERAL_CHANNEL_ID;
 const USERNAME_REGEX = /<@(\d+)>/g;
 const TODO_REGEX = /TODO/;
+const TALKING_POINT_REGEX = /(Talking point|TP)/;
```
Comment (rajpiskala) [inline index.js]: Keep in mind this is case-sensitive. If it should be case insensitive, then you can add an `i` past the `/` in the regex to make it case insensitive

File: index.js [hunk 2]
```diff
@@ -33,9 +33,11 @@ const TALKING_POINT_TARGET_CHANNEL_ID = process.env.TALKING_POINT_CHANNEL_ID;
...
 const TIME_IN_DAYS = 1000 * 60 * 60 * 24;
 const CHAR_LENGTH = 500;
 const deleteTodo = [false, false];
+let talkingPointCount = 0;
```
Comment (rajpiskala) [inline index.js]: Is there a way we could have it pick up the talking point count even after the bot restarts? One way would be to check the latest message in the Talking Points channel, read its title, and get the count from that

File: index.js [hunk 3]
```diff
@@ -176,6 +178,65 @@ client.on("messageCreate", async (message) => {
...
+        try {
+            // Prompt user for title:
+            await message.reply("Please provide a title for this TP:");
+
+            // Create a message collector for the title:
+            const filter = (response) => {
+                return response.author.id === message.author.id;
+            };
+
+            const collector = new MessageCollector(message.channel, { time: 60000, max: 1, filter: filter });
```
Comment (rajpiskala) [inline index.js]: Please make 60000 into a constant

**Review by rajpiskala — APPROVED** [review]
Change looks good, feel free to merge after addressing comments

File: index.js (hunk 1 above)
Comment (ThePhysic) [reply index.js]: Ack

File: index.js (hunk 2 above)
Comment (ThePhysic) [reply index.js]: That's definitely doable

File: index.js (hunk 3 above)
Comment (ThePhysic) [reply index.js]: Nice catch again

**Pull request merged**